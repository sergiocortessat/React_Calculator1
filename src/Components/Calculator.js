import { useState, useEffect } from 'react';
import Calculate from '../Logic/calculate';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import logic from '../Logic/DisplayLogic';

const Calculator = () => {
  const [total, setTotal] = useState(null);
  const [operation, setOperation] = useState(null);
  const [next, setNext] = useState(null);
  const [result, setResult] = useState(null);
  const handleClick = (buttonName) => {
    const data = { total, operation, next };
    const result = Calculate(data, buttonName);
    setTotal(result.total);
    setOperation(result.operation);
    setNext(result.next);
  };

  useEffect(() => {
    const resultLogic = logic({ total, next, operation });
    setResult(resultLogic);
  }, [result, total, next, operation]);

  return (
    <>
      <div className="app">
        <h4 className="title-2 text-center">The Matrix Calculator</h4>
        <div>
          <Display result={result} />
          <ButtonPanel clickHandler={handleClick} />
        </div>
      </div>
    </>
  );
};
export default Calculator;
