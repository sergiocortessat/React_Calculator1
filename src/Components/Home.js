import React from 'react';
import { Route, Switch } from 'react-router-dom';
import NavBar from './NavBar';
import HomeMessage from './HomeMessage';
import '../style/HomeTitle.scss';

function Home() {
  return (
    <>
      <NavBar />
      <h2 className="title">Welcome to math magicians</h2>
      <Switch>
        <Route exact path="/" component={HomeMessage} />
      </Switch>
    </>
  );
}

export default Home;
