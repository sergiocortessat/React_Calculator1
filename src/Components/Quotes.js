import React from 'react';
import style from '../style/Quotes.module.scss';

function Quotes() {
  return (
    <div className={style.quotes} data-testid="quote">
      <h2>Unfortunately, no one can be told what the Matrix is. You have to see it for yourself</h2>
      <p>Morpheus</p>
    </div>
  );
}

export default Quotes;
