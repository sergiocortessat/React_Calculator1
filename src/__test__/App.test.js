import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../Components/Home';
import Calculator from '../Components/Calculator';
import Quotes from '../Components/Quotes';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <>
        <BrowserRouter>
          <Home />
          <Switch>
            <Route path="/calculator" component={Calculator} />
            <Route path="/quotes" component={Quotes} />
          </Switch>
        </BrowserRouter>
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
