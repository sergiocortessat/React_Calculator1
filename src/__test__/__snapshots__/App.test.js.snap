// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav>
    <ul
      className="navbar"
    >
      <li>
        <a
          aria-current="page"
          className="current"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home 
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator 
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/quotes"
          onClick={[Function]}
        >
          Quotes 
        </a>
      </li>
    </ul>
  </nav>,
  <h2
    className="title"
  >
    Welcome to math magicians
  </h2>,
  <h1
    className="home-message"
    data-testid="home-messages"
  >
    Welcome back Neo. We were expecting you back.
  </h1>,
]
`;
